gems/sorbet-runtime/test/types/final_method.rb

```diff
@@ -184,6 +184,42 @@ def bar; end
     end
   end
 
+  it "allows declaring a final instance method and a final class method with the same name" do
+    c = Class.new do
+      extend T::Sig
+      sig(:final) {returns(Symbol)}
+      def foo
+        :instance
+      end
+
+      sig(:final) {returns(Symbol)}
+      def self.foo
+        :class
+      end
+    end
+
+    assert_equal(:instance, c.new.foo)
+    assert_equal(:class, c.foo)
+  end
+
+  it "allows declaring a final class method and a non-final instance method with the same name" do
+    c = Class.new do
+      extend T::Sig
+      sig(:final) {returns(Symbol)}
+      def self.foo
+        :class
+      end
+
+      sig {returns(Symbol)}
+      def foo
+        :instance
+      end
+    end
+
+    assert_equal(:instance, c.new.foo)
+    assert_equal(:class, c.foo)
+  end
```

Comment:
Does it make sense to add a test for a final instance method, and non-final class method as well, just to complete the enumeration of cases?

Reply:
Yup, done!